common/src/View/VertexToolBase.h

```diff
@@ -231,7 +234,32 @@ namespace TrenchBroom {
                 m_dragging = false;
                 m_ignoreChangeNotifications.popLiteral();
             }
-            
+
+        public: // csg convex merge
+            bool canDoCsgConvexMerge() {
+                return handleManager().selectedHandleCount() > 1;
+            }
+
+            void csgConvexMerge() {
+                std::vector<vm::vec3> vertices;
+                const auto handles = handleManager().selectedHandles();
```

Comment:
I don't think that this is what you want, right? For the edge and face vertex tool, the handles are `vm::segment3` and `vm::polygon3`, respectively, I think. I am surprised that you don't get a type error. What you want is to collect the adjacent vertices in those cases and not the handles themselves. I think you need to call a virtual method that is implemented in the respective tools. This method should then obtain the positions of the vertices which are adjacent to the selected handles.

Reply:
For edges and faces, handles is indeed a vector<segment3> or vector<polygon3>, respectively. But then I pass that to the static method getVertices that exists in both the segment and polygon classes. It collects all vertices from a range of segments or polygons, so the vector<vec3> vertices here does get filled will all vertices adjacent to the selected handles and not the handles themselves.

Reply:
Nice use of static polymorphism! I like it.